src/main/java/com/kodilla/ecommercee/services/UserService.java

```diff
@@ -0,0 +1,34 @@
+package com.kodilla.ecommercee.services;
+
+import com.kodilla.ecommercee.domains.User;
+import com.kodilla.ecommercee.exceptions.UserNotFoundException;
+import com.kodilla.ecommercee.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.Random;
+
+
+@Service
+public class UserService {
+    @Autowired
+    private UserRepository repository;
+
+    public User createUser(final User user) {
+        return repository.save(user);
+    }
+
+    public User blockUserById(final Long userId) throws UserNotFoundException {
+        User blockUser = repository.findById(userId).orElseThrow(UserNotFoundException::new);
+        blockUser.setStatus(false);
+        return repository.save(blockUser);
+    }
+
+    public User generateKey(final String username) throws UserNotFoundException {
+        Random rand = new Random();
+        long key = rand.nextInt(90000) + 10000;
+
+        User generateNumber = repository.findUserByUsername(username).orElseThrow(UserNotFoundException::new);
```

Comment:
nazwa zmiennej wskazuje że będzie to jakaś wartość liczbowa a nie User, moim zdaniem do zmiany.

Reply:
Klucz jest przypisywany do uzytkownika po dodaniu przez niego wartosci "username", wiec klucz jest automatycznie przypisany i uzytkownik jest uaktualniony.

Reply:
Aaaa, ok, zmienione :)